tutorial/gui/pcl-visualizer/tutorial-pcl-visualizer.cpp

```diff
@@ -0,0 +1,205 @@
+      i+=2;
+    }
+    else if (std::string(argv[i]) == "--reso" && i + 2 < argc)
+    {
+      opt_reso.first = atoi(argv[i + 1]);
+      opt_reso.second = atoi(argv[i + 2]);
+      i+=2;
+    }
+    else if (std::string(argv[i]) == "--display-mode" && i + 1 < argc)
+    {
+      opt_mode = displayModeFromString(std::string(argv[i + 1]));
+      i++;
+    }
+    else if (std::string(argv[i]) == "--help" || std::string(argv[i]) == "-h")
+    {
+      std::cout << "NAME" << std::endl;
+      std::cout << "\t" << argv[0] << "  Test programm for the PCL-based point-cloud visualizer." << std::endl
+                << std::endl;
+      std::cout << "SYNOPSIS" << std::endl;
+      std::cout << "\t" << argv[0]
+                << "\t[--noise <stdev_noise>] (default: " + std::to_string(def_addedNoise)<< ")\n"
+                << "\t[--order <surface-order>](default: " + std::to_string(def_order) << ")\n"
+                << "\t[--x-lim <xmin xmax>](default: [" + std::to_string(def_xlim.first) + ";" + std::to_string(def_xlim.second) << "])\n"
+                << "\t[--y-lim <ymin ymax>](default: [" + std::to_string(def_ylim.first) + ";" + std::to_string(def_ylim.second) << "])\n"
+                << "\t[--reso <x_resolution y_resolution>](default: [" + std::to_string(def_reso.first) + ";" + std::to_string(def_reso.second) << "])\n"
+                << "\t[--display-mode " << getAvailableDisplayMode() <<"](default: " << displayModeToString(def_mode) << ")\n"
+                << "\t[--help] [-h]" << std::endl
+                << std::endl;
+
+      return EXIT_SUCCESS;
+    }
+  }
+  //! [Arguments parser]
+
+  std::cout << "Parameters:" << std::endl;
+  std::cout << "\tSurface order: " << opt_order << std::endl;
+  std::cout << "\tX-axis limits: [" << opt_xlim.first << " ; " << opt_xlim.first << "]" << std::endl;
+  std::cout << "\tY-axis limits: [" << opt_ylim.first << " ; " << opt_ylim.first << "]" << std::endl;
+  std::cout << "\tGrid resolution: [" << opt_reso.first << " x " << opt_reso.first << "]" << std::endl;
+  std::cout << "\tNoise standard deviation: " << opt_addedNoise << std::endl;
+  std::cout << "\tDisplay mode: " << displayModeToString(opt_mode) << std::endl;
+
+  //! [Running blocking mode]
+  if(opt_mode == DisplayMode::BLOCKING || opt_mode == DisplayMode::BOTH)
+  {
+    ClassUsingPclVisualizer demo(opt_xlim, opt_ylim, opt_reso);
+    demo.blockingMode(opt_addedNoise, opt_order);
+  }
+  //! [Running blocking mode]
+
+  //! [Running threaded mode]
+  if(opt_mode == DisplayMode::THREADED || opt_mode == DisplayMode::BOTH)
+  {
+    ClassUsingPclVisualizer demo(opt_xlim, opt_ylim, opt_reso);
+    demo.threadedMode(opt_addedNoise, opt_order);
+  }
+  //! [Running threaded mode]
+
+  return 0;
+}
+#else
+
+int main(int argc, char *argv[])
+{
+  std::cerr << "ERROR: ViSP seems to have been compiled without PCL library" << std::endl;
+  std::cerr << "\tPlease install PCL library and recompile ViSP." << std::endl;
+}
+#endif
```

Comment:
Add an empty line

Reply:
Done in [91391b3](https://github.com/lagadic/visp/pull/1163/commits/91391b324ed48da41cfb5034f2807dc2e77fa549)